src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,220 @@
+    if (compositeBoxesToParse.indexOf(boxtype) >= 0) {
+      output = boxSpecificData.subarray(0, 8);
+      let pos = 8;
+      while (pos < boxSpecificData.length) {
+        const {
+          size: childSize,
+          boxtype: childBoxType,
+        } = getBoxInfos(boxSpecificData.subarray(pos, pos + 8));
+        const outputChildBox =
+          patchBox(
+            childBoxType,
+            boxSpecificData.subarray(pos, pos + childSize),
+            filePos + pos,
+            path
+          );
+        output = concat(output, outputChildBox);
+        pos += childSize;
+      }
+      if (output.length !== boxSpecificData.length) {
+        output = concat(itobe4(output.length), output.subarray(4, output.length));
+      }
+    } else {
+      switch (boxtype) {
+        case "styp":
+          output = patchStyp(boxSpecificData);
+          break;
+        case "trun":
+          output = patchTrun(boxSpecificData);
+          break;
+        case "tfdt":
+          output = patchTfdt(boxSpecificData);
+          break;
+        default:
+          output = boxSpecificData;
+          break;
+      }
+    }
+    return output;
+  }
+
+  /**
+   * Process styp and make sure lmsg presence follows the lmsg flag parameter.
+   * @param {Uint8Array} input
+   * @return {Uint8Array} patched box
+   */
+  function patchStyp(input: Uint8Array): Uint8Array {
+    const size = be4toi(input, 0);
+    let pos = 8;
+    const brands = [];
+    while (pos < size) {
+        const brand = input.subarray(pos, pos + 4);
+        if (bytesToStr(brand) !== "lmsg") {
+            brands.push(brand);
+        }
+        pos += 4;
+    }
+    if (lmsg) {
+      brands.push(strToBytes("lmsg"));
+    }
+    const dataSize = brands.length * 4;
+    const stypData = new Uint8Array(dataSize);
+    for (let i = 0; i < brands.length; i++) {
+      stypData.set(brands[i], i * 4);
+    }
+    return Atom("styp", stypData);
+  }
+
+  function patchTrun(input: Uint8Array): Uint8Array {
+    const flags = be4toi(input, 8) & 0xFFFFFF;
+    let dataOffsetPresent = false;
+    // Data offset present
+    if (flags & 0x1) {
+      dataOffsetPresent = true;
+    }
+    // Modify data_offset
+    let output = input.subarray(0, 16);
+    if (dataOffsetPresent && sizeChange > 0) {
+      let trunOffset = be4toi(input, 16);
+      trunOffset += sizeChange;
+      output = concat(output, itobe4(trunOffset)); // func
+    }
+    else {
+      output = concat(output, input.subarray(16, 20));
+    }
+    output = concat(output, input.subarray(20, input.length));
+    return output;
+  }
+
+  /**
+   * Generate new timestamps for tfdt and change size of boxes above if needed.
+   * Try to keep in 32 bits if possible.
+   * @param input
+   */
+  function patchTfdt(input: Uint8Array): Uint8Array {
+    const version = input[8];
+    const tfdtOffset = timeOffset;
+    let newBaseMediaDecodeTime;
+    let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWLjl6y--pl6AHs2a-uS&open=AWLjl6y--pl6AHs2a-uS)

[ISSUE_KEY]: AWLjl6y--pl6AHs2a-uS